hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/OzoneClientAdapter.java

```diff
@@ -46,9 +46,17 @@ public interface OzoneClientAdapter {
   OzoneFSOutputStream createFile(String key, short replication,
       boolean overWrite, boolean recursive) throws IOException;
 
+  OzoneFSOutputStream createFile(String key, short replication,
+      boolean overWrite, boolean recursive,
+      String ecPolicyName) throws IOException;
```

Comment:
@sodonnel My idea is to keep all the previous parameters and only add an ecPolicyName attribute. Maybe the previous parameters will be useful later. Forcibly convert fs to RootedOzoneFileSystem in hadoop, and then call the createFile method, which is similar to forcibly converting fs to DistributedFileSystem above.

Reply:
> Forcibly convert fs to RootedOzoneFileSystem in hadoop ... similar to forcibly converting fs to DistributedFileSystem

I don't think this will work.  Hadoop has no dependency on Ozone.

Reply:
@adoroszlai At present, all I can think of is to add an interface in the ozone, and then pass the ecPolicyName attribute to the ozone on the hadoop side. Do you think there are other solutions?

Reply:
@sodonnel Thank you for your reply.

In response to this question The data is EC in Ozone, but goes to a non EC folder in HDFS - how will that work, and what will the replication factor be set to in HDFS? The result of my test is 3.

At this jira https://issues.apache.org/jira/browse/HADOOP-14254  add a "-pe" switch to ensure erasure coded files at source are copied as erasure coded files at destination.

But when I distcp specify -p HDFS EC to Ozone Ratis Bucket, The number of replicas in ozone is neither ec nor RATIS-THREE, but Ratis-ONE.

In this pr https://github.com/apache/hadoop/commit/c757cb61ebc9e69d9f6f143da91189b9f0517ee9#diff-6207b5e70af95b856263703af631c577358dd06662d364f06347827d92f734f2. His conditions are very harsh, `sourceStatus instanceof HdfsFileStatus` and `targetFS instanceof DistributedFileSystem`
Only then will the ecPolicy attribute be obtained, which causes distcp to cannot obtain the ecPolicy in the ozone.

![image](https://github.com/apache/ozone/assets/35120306/42d4dc02-5453-4c53-88a5-21dca4f6786e)
![image](https://github.com/apache/ozone/assets/35120306/27b8e521-6de9-43bf-a677-9467b93bd479)

Reply:
I see that https://issues.apache.org/jira/browse/HDFS-10996 added a "per file" EC setting to the `hadoop-hdfs-project/hadoop-hdfs-client/src/main/java/org/apache/hadoop/hdfs/DistributedFileSystem.java` class - does Ozone already extend this class, and if so, perhaps the HDFS interface is already is place.

Reply:
@sodonnel As far as I understand, ozone is not extend DistributedFileSystem.

Reply:
Ozone extends FileSystem, and it looks like distcp either talks to a DistributedFileSystem (HDFS) or a FileSystem. Therefore I suspect we need to change the FileSystem class in Hadoop to have the new parameter first. Otherwise, even though we have added the new parameters here, distcp will not be able to see them, as it only knows about Filesystem. We cannot cast an object in distcp into an Ozone class.

Reply:
> Ozone extends FileSystem, and it looks like distcp either talks to a DistributedFileSystem (HDFS) or a FileSystem. Therefore I suspect we need to change the FileSystem class in Hadoop to have the new parameter first. Otherwise, even though we have added the new parameters here, distcp will not be able to see them, as it only knows about Filesystem. We cannot cast an object in distcp into an Ozone class.

@sodonnel I think this is possible, cc @ChenSammi